src/hotspot/share/services/memoryPool.cpp

```diff
@@ -184,6 +184,8 @@ MemoryUsage CodeHeapPool::get_memory_usage() {
   return MemoryUsage(initial_size(), used, committed, maxSize);
 }
 
+// Note, any NonHeap pools would be counted into Non-Heap memory by MemoryMXBean.getNonHeapMemoryUsage.
+// Make sure that it behaves as you expected if you want to add a new type of NonHeap pool
 MetaspacePool::MetaspacePool() :
```

Comment:
Can you please add a comment like
"Accounts for Metaspace (both class space and non-class space).
 Max usage is MaxMetaspaceSize. Current usage includes prematurely freed blocks.
 As far as this API is concerned, CompressedClassSpaceSize is ignored."